compiler-core/src/error.rs

```diff
@@ -1879,19 +1881,49 @@ But function expects:
                     text.push_str(&printer.print_type(expected));
                     text.push_str("\n\nFound type:\n\n    ");
                     text.push_str(&printer.print_type(given));
+
+                    let (main_message_location, main_message_text, extra_labels) = match situation {
+                        // When the mismatch error comes from a case clause we want to highlight the
+                        // entire branch (pattern included) when reporting the error; in addition,
+                        // if the error could be resolved just by wrapping the value in an `Ok`
+                        // or `Error` we want to add an additional label with this hint below the
+                        // offending value.
+                        Some(UnifyErrorSituation::CaseClauseMismatch { clause_location }) =>
+                            match hint_wrap_value_into_result(expected, given) {
+                                None => (clause_location, None, vec![]),
+                                Some(hint) => (
+                                    clause_location,
+                                    None,
+                                    vec![
+                                        ExtraLabel {
+                                            src_info: None,
+                                            label: Label {
+                                                text: Some(hint),
+                                                span: *location,
+                                            }
+                                        },
```

Comment:
I'm finding this quite hard to read with all the nesting, it's easy to miss that it's returning a tuple. Could you assign this bit to a variable pleaes